.github/workflows/benchmark.yml

```diff
@@ -0,0 +1,191 @@
+        AZURE_RELAY: ${{ secrets.AZURE_RELAY }}
+      if: ${{ steps.benchmark-argument.outputs.result == 'kestrel' }}
+      run: |
+        CRANK_PROFILE="aspnet-perf-lin-relay"
+        CRANK_CONFIGS="--config https://github.com/aspnet/Benchmarks/blob/main/build/azure.profile.yml?raw=true "
+        CRANK_ARGUMENTS="--config https://raw.githubusercontent.com/aspnet/Benchmarks/main/scenarios/plaintext.benchmarks.yml --scenario plaintext --application.framework net6.0 --application.options.outputFiles ./artifacts/bin/Microsoft.AspNetCore.Server.Kestrel/release/net6.0/"
+
+        git checkout $GITHUB_BASE_REF
+
+        echo "Building Kestrel"
+        ./src/Servers/Kestrel/build.sh -c release
+
+        crank $CRANK_CONFIGS --profile $CRANK_PROFILE --relay AZURE_RELAY --json "$GITHUB_WORKSPACE/base.json" $CRANK_ARGUMENTS
+
+        git fetch origin pull/${{ github.event.issue.number }}/head
+        git config --global user.name "user"
+        git config --global user.email "user@company.com"
+        git merge FETCH_HEAD
+
+        echo "Building Kestrel"
+        ./src/Servers/Kestrel/build.sh -c release
+
+        crank $CRANK_CONFIGS --profile $CRANK_PROFILE --relay AZURE_RELAY --json "$GITHUB_WORKSPACE/head.json" $CRANK_ARGUMENTS
+
+        cd $GITHUB_WORKSPACE
+        echo "Arguments:"$'\n```\n'$"$CRANK_ARGUMENTS"$'\n```\n' >> results.txt
+        echo "Results:"$'\n```\n' >> results.txt
+        crank compare base.json head.json >> results.txt
+        echo $'\n```\n' >> results.txt
+
+        cat results.txt
+
+    - name: Benchmark Mvc
+      working-directory: ./aspnetcore
+      env:
+        AZURE_RELAY: ${{ secrets.AZURE_RELAY }}
+      if: ${{ steps.benchmark-argument.outputs.result == 'mvc' }}
+      run: |
+        CRANK_PROFILE="aspnet-perf-lin-relay"
+        CRANK_CONFIGS="--config https://github.com/aspnet/Benchmarks/blob/main/build/azure.profile.yml?raw=true "
+        CRANK_ARGUMENTS="--config https://raw.githubusercontent.com/aspnet/Benchmarks/main/scenarios/plaintext.benchmarks.yml --scenario mvc --application.framework net6.0 --application.options.outputFiles ./artifacts/bin/Microsoft.AspNetCore.Mvc.IntegrationTests/release/net6.0/"
+
+        git checkout $GITHUB_BASE_REF
+
+        echo "Building Mvc"
+        ./src/Mvc/build.sh -c release -NoBuildNodeJS
+
+        crank $CRANK_CONFIGS --profile $CRANK_PROFILE --relay AZURE_RELAY --json "$GITHUB_WORKSPACE/base.json" $CRANK_ARGUMENTS
+
+        git fetch origin pull/${{ github.event.issue.number }}/head
+        git config --global user.name "user"
+        git config --global user.email "user@company.com"
+        git merge FETCH_HEAD
+
+        echo "Building Mvc"
+        ./src/Mvc/build.sh -c release -NoBuildNodeJS
+
+        crank $CRANK_CONFIGS --profile $CRANK_PROFILE --relay AZURE_RELAY --json "$GITHUB_WORKSPACE/head.json" $CRANK_ARGUMENTS
+
+        cd $GITHUB_WORKSPACE
+        echo "Arguments:"$'\n```\n'$"$CRANK_ARGUMENTS"$'\n```\n' >> results.txt
```

Comment:
Should this bit be put in a separate step so it's used by all benchmarks?

Reply:
This is intentional as I expect some commands to diverge from this and call crank multiple times, or generate an output that is different. So just repeating to simplify authoring new commands.